homeassistant/components/august/__init__.py

```diff
@@ -75,138 +57,154 @@
     try:
-        authentication = authenticator.authenticate()
-    except RequestException as ex:
-        _LOGGER.error("Unable to connect to August service: %s", str(ex))
-
-        hass.components.persistent_notification.create(
-            "Error: {ex}<br />You will need to restart hass after fixing.",
-            title=NOTIFICATION_TITLE,
-            notification_id=NOTIFICATION_ID,
-        )
+        august_gateway.authenticate()
+    except RequireValidation:
+        await async_request_validation(hass, config_entry, august_gateway)
+        return False
+    except InvalidAuth:
+        _LOGGER.error("Password is no longer valid. Please set up August again")
+        return False
 
-    state = authentication.state
+    entry_id = config_entry.entry_id
+    # We still use the configurator to get a new 2fa code
+    # when needed since config_flow doesn't have a way
+    # to re-request if it expires
+    if entry_id in _CONFIGURING:
+        hass.components.configurator.async_request_done(_CONFIGURING.pop(entry_id))
 
-    if state == AuthenticationState.AUTHENTICATED:
-        if DOMAIN in _CONFIGURING:
-            hass.components.configurator.request_done(_CONFIGURING.pop(DOMAIN))
+    hass.data[DOMAIN][config_entry.entry_id] = await hass.async_add_executor_job(
+        AugustData, hass, august_gateway
+    )
 
-        hass.data[DATA_AUGUST] = AugustData(
-            hass, api, authentication, authenticator, token_refresh_lock
+    for component in AUGUST_COMPONENTS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(config_entry, component)
         )
 
-        for component in AUGUST_COMPONENTS:
-            discovery.load_platform(hass, component, DOMAIN, {}, config)
+    return True
 
-        return True
-    if state == AuthenticationState.BAD_PASSWORD:
-        _LOGGER.error("Invalid password provided")
-        return False
-    if state == AuthenticationState.REQUIRES_VALIDATION:
-        request_configuration(hass, config, api, authenticator, token_refresh_lock)
-        return True
 
-    return False
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the August component from YAML."""
 
+    conf = config.get(DOMAIN)
 
-async def async_setup(hass, config):
-    """Set up the August component."""
+    if not conf:
+        return True
 
-    conf = config[DOMAIN]
-    api_http_session = None
-    try:
-        api_http_session = Session()
-    except RequestException as ex:
-        _LOGGER.warning("Creating HTTP session failed with: %s", str(ex))
-
-    api = Api(timeout=conf.get(CONF_TIMEOUT), http_session=api_http_session)
-
-    authenticator = Authenticator(
-        api,
-        conf.get(CONF_LOGIN_METHOD),
-        conf.get(CONF_USERNAME),
-        conf.get(CONF_PASSWORD),
-        install_id=conf.get(CONF_INSTALL_ID),
-        access_token_cache_file=hass.config.path(AUGUST_CONFIG_FILE),
+    hass.data.setdefault(DOMAIN, {})
```

Comment:
This should be before the `return True` above here

Reply:
done